drivers/sensor/melexis/MLX90394/mlx90394.c

```diff
@@ -0,0 +1,601 @@
+{
+	int rt;
+	struct mlx90394_data *data = dev->data;
+
+	rt = mlx90394_trigger_measurement_internal(dev, chan);
+	if (rt != 0) {
+		return rt;
+	}
+	k_usleep(data->measurement_time_us);
+	rt = mlx90394_sample_fetch_internal(dev, chan);
+	if (rt != 0) {
+		return rt;
+	}
+	return 0;
+}
+
+static void mlx90394_calc_measurement_Time_us(struct mlx90394_data *data)
+{
+	int32_t enX = FIELD_GET(MLX90394_CTRL1_X_EN, data->ctrl_reg_values.ctrl1);
+	int32_t enY = FIELD_GET(MLX90394_CTRL1_Y_EN, data->ctrl_reg_values.ctrl1);
+	int32_t enZ = FIELD_GET(MLX90394_CTRL1_Z_EN, data->ctrl_reg_values.ctrl1);
+	int32_t enTemp = FIELD_GET(MLX90394_CTRL4_T_EN, data->ctrl_reg_values.ctrl4);
+	int32_t filterHallXY =
+		FIELD_GET(MLX90394_CTRL3_DIG_FILT_HALL_XY, data->ctrl_reg_values.ctrl3);
+	int32_t filterHallZ =
+		FIELD_GET(MLX90394_CTRL4_DIG_FILT_HALL_Z, data->ctrl_reg_values.ctrl4);
+	int32_t filterTemp = FIELD_GET(MLX90394_CTRL3_DIG_FILT_TEMP, data->ctrl_reg_values.ctrl3);
+	int32_t osrTemp = FIELD_GET(MLX90394_CTRL3_OSR_TEMP, data->ctrl_reg_values.ctrl3);
+	int32_t osrHall = FIELD_GET(MLX90394_CTRL3_OSR_HALL, data->ctrl_reg_values.ctrl3);
+
+	int32_t conversion_time_us =
+		(osrHall + 1) * ((enX + enY) * MLX90394_CONVERSION_TIME_US_AXIS[filterHallXY] +
+				 enZ * MLX90394_CONVERSION_TIME_US_AXIS[filterHallZ]) +
+		(osrTemp + 1) * enTemp * MLX90394_CONVERSION_TIME_US_AXIS[filterTemp];
+	int32_t dsp_time_us = MLX90394_DSP_TIME_US[enTemp][enX + enY + enZ];
+
+	/*
+	 * adding 5% tollerance from datasheet
+	 */
+	data->measurement_time_us = (conversion_time_us + dsp_time_us) * 105 / 100;
+}
+
+static int mlx90394_init(const struct device *dev)
+{
+	const struct mlx90394_config *cfg = dev->config;
+	struct mlx90394_data *data = dev->data;
+	int rc;
+
+	if (!i2c_is_ready_dt(&cfg->i2c)) {
+		LOG_ERR("I2C bus device not ready");
+		return -ENODEV;
+	}
+
+	/*
+	 * Soft reset the chip
+	 */
+	rc = i2c_reg_write_byte_dt(&cfg->i2c, MLX90394_REG_RST, MLX90394_RST);
+	if (rc != 0) {
+		LOG_ERR("Failed to soft reset");
+		return -EIO;
+	}
+	k_usleep(MLX90394_STARTUP_TIME_US);
+
+	/*
+	 * check chip ID
+	 */
+	rc = mlx90394_check_who_am_i(&cfg->i2c);
+	if (rc != 0) {
+		return rc;
+	}
+
+	/*
+	 * set all to default and read the settings back
+	 */
+	rc = mlx90394_write_read_dt(&cfg->i2c, MLX90394_REG_CTRL1,
+				    (uint8_t *)&data->ctrl_reg_values.ctrl1,
+				    (uint8_t *)&data->ctrl_reg_values.ctrl1, 2);
+	if (rc != 0) {
+		return rc;
+	}
+
+	rc = mlx90394_write_read_dt(&cfg->i2c, MLX90394_REG_CTRL3,
+				    (uint8_t *)&data->ctrl_reg_values.ctrl3,
+				    (uint8_t *)&data->ctrl_reg_values.ctrl3, 2);
+	if (rc != 0) {
+		return rc;
+	}
+
+	mlx90394_calc_measurement_Time_us(data);
+
+#ifdef CONFIG_SENSOR_ASYNC_API
+	data->dev = dev;
+	/*
+	 * init work for fetching after measurement has completed
+	 */
+	k_work_init_delayable(&data->async_fetch_work, mlx90394_async_fetch);
+#endif
+
+	return 0;
+}
+
+static const struct sensor_driver_api mlx90394_driver_api = {
```

Comment:
Device drivers APIs need to be wrapped in `DEVICE_API` macro.

```suggestion
static DEVICE_API(sensor, mlx90394_driver_api) = {
```